### ISSUE TITLE
Chore: Validate and Trim Spaces for All User Input Fields

### ISSUE THREAD

**Issue by bupd** [goharbor/harbor-cli#195]
## Expected Outcome
- All input fields should properly validate and trim spaces.
- If an input field contains only spaces or empty it should trigger an appropriate error message.

You can make use of Validation method mentioned [here.](https://github.com/goharbor/harbor-cli/pull/66#pullrequestreview-2312492192)

**Comment by JianMinTang** [comment]
Hi, @bupd, I'm interested in this issue and have two ideas to solve the input problem. 
I'm not sure which one is better. 
The first one is:
![harborcli-#195-1](https://github.com/user-attachments/assets/7c7a856e-4ff1-4099-8f56-17ddbdcb5d43)
The second one is: 
![harborcli-#195-2](https://github.com/user-attachments/assets/537fac7a-6a5e-48e0-a6dc-5ce511669592)
And I'm not clear about the rules for trimming the string of input, do I need to trim only the leading of spaces, only the trailing of spaces, both the leading and trailing of spaces? Or are there other rules?
I'd really appreciate any additional helpful ideas you have to share!

**Comment by bupd** [comment]
Hello @JianMinTang, thanks for your interest in addressing this issue.
> And I'm not clear about the rules for trimming the string of input, do I need to trim only the leading of spaces, only the trailing of spaces, both the leading and trailing of spaces? Or are there other rules?

Regarding your question on trimming the input string: 
The Go method [`strings.TrimSpace`](https://pkg.go.dev/strings#TrimSpace) will indeed trim both leading and trailing whitespaces. This should work well for cleaning up the input.

Here's my suggested approach to handle this across all user input fields:

- Step 1: Use `strings.TrimSpace` to remove both leading and trailing spaces from the input.
- Step 2: Check if the resulting string is empty, and if so, return an appropriate error.
- Additional Step: You can also apply more formatting rules & appropriate error messages as needed for specific fields to ensure the input adheres to the expected API resource model.

For reference, please take a look at the implementation in [this PR review](https://github.com/goharbor/harbor-cli/pull/66#pullrequestreview-2312492192). You can replicate the pattern used there for all relevant input fields. That's exactly the kind of input validation we need to implement throughout.

Feel free to share any further thoughts or ideas you might have!

**Comment by JianMinTang** [comment]
Hello @bupd, i appreciate your encouraging and helpful reply in this issue, i try to implement the pattern code.
```
huh.NewInput().
            Title("Server").
            Value(&loginView.Server).
            Validate(func(str string) error {
	            trimmedStr := strings.TrimSpace(str)
	            loginView.Server = trimmedStr
	            if trimmedStr == "" {
		            return errors.New("server cannot be empty or only spaces")
	            }
	            return nil
            }),
```
But the input by user in console just store in `loginView.Server`. So the method `Validate` just check format of the **`str`** and the formatted string(`trimmedStr`) cannot be assigned to `loginView.Server`.
**My idea is to ensure the validity of the string when the user inputs it in the console.**

**Comment by JianMinTang** [comment]
Hi @bupd, I'm interested in the issue related to the merged PR #198.

**Comment by bupd** [comment]
Closing #195 as completed via #198
